api/util.py

```diff
@@ -88,11 +88,12 @@ def path_from_hash(hash_):
     """
     create a filepath from a hash
     e.g.
-    hash_ = 01b395a1cbc0f218
+    hash_ = v0-sha384-01b395a1cbc0f218
     will return
-    0/1/b/3/9/5/a/1/01b395a1cbc0f218
+    v0/sha384/01/b3/v0-sha384-01b395a1cbc0f218
     """
-    path = [hash_[i] for i in xrange(8)]
+    hash_version, hash_alg, actual_hash = hash_.split('-')
+    path = [hash_version, hash_alg] + [actual_hash[2 * i : 2 * i + 2] for i in range(2)]
```

Comment:
Regarding `[actual_hash[2 * i : 2 * i + 2] for i in range(2)]`:

I really want us to get away from inline array building.
This instance requires mental reverse-engineering to understand it's selecting two sets of two characters.

How about `[actual_hash[0:2], actual_hash[3:5]]` instead?
Or even better:

```python
first_stanza = actual_hash[0:2]
second_stanza = actual_hash[3:5]
path = [hash_version, hash_alg, first_stanza, second_stanza]
```